typescript/testsuite/testsuite_impl/advanced_network_test/advanced_network_test_.ts

```diff
@@ -0,0 +1,75 @@
+import { NetworkContext, Network } from "kurtosis-core-api-lib";
+import { TestNetwork } from "../../network_impl/test_network"
+import { TestConfigurationBuilder } from "kurtosis-testsuite-api-lib"; //TODO (Ali)
+import * as log from "loglevel";
+import { ok, err, Result } from "neverthrow";
+import { APIClient, Person } from "../../../api/api_service_client/api_client";
+
+const TEST_PERSON_ID: number = 46;
+
+export class AdvancedNetworkTest {
+	private readonly datastoreServiceImage: string;
+	private readonly apiServiceImage: string;
+	
+	constructor (datastoreServiceImage: string, apiServiceImage: string) {
+		this.datastoreServiceImage = datastoreServiceImage; 
+		this.apiServiceImage = apiServiceImage;
+	}
+
+	public configure(builder: TestConfigurationBuilder): void {
+		builder.withSetupTimeoutSeconds(60).withRunTimeoutSeconds(60); //TODO (Ali) - allowed since typescript gives direct reference
+	}
+
+	public async setup(networkCtx: NetworkContext): Promise<Result<Network, Error>> { //TODO (Ali) - allowed to be async
```

Comment:
Yep, this should be `async` so that the user can write code as easily as possible in a test